designs/2022-notify-plugin-on-fix/README.md

```diff
@@ -0,0 +1,151 @@
+-   Repo: eslint/eslint
+-   Start Date: 2022-09-19
+-   RFC PR: (leave this empty, to be filled in later)
+-   Authors: [Jason Dent](https://github.com/jason3s)
+
+# Notify Plugins on Fix
+
+## Summary
+
+Notify Plugins when a fix has been applied. Currently a plugin has no way to know if a fix has been applied.
+
+## Motivation
+
+The primary motivation for this RFC is to support interactive environments such as an editor where the actual fix might be an external side-effect.
+
+### Context
+
+As the author of the spellchecker plugin: [@cspell/eslint-plugin](https://www.npmjs.com/package/@cspell/eslint-plugin), I would like to present the user a "fix" that allows the user to add a word to their custom dictionary. See: [CSpell ESLint Plugin: Configuration option to add words to the dictionary · Issue #3233](https://github.com/streetsidesoftware/cspell/issues/3233).
+
+When used with an editor extension like VS Code ESLint, something like the following would show up:
+
+<img width="402" alt="image" src="./quick-fix.png">
+
+If the fix is applied, the spellchecker plugin would add the word to the custom dictionary.
```

Comment:
VS Code applies individual autofixes and suggestions independently of ESLint, based on the provided [`EditInfo`](https://eslint.org/docs/latest/developer-guide/nodejs-api#-editinfo-type). ESLint doesn't know if the fix/suggestion is applied in the editor. It seems that we should provide an API for editors to notify ESLint when the fix/suggestion has been applied.

Reply:
> It's not clear to me whether this feature relates to autofixes or [suggestions](https://eslint.org/docs/latest/developer-guide/working-with-rules#providing-suggestions) or both. 

In the spell checker case, it is explicitly for `suggestions` over `autofixes`.

Reply:
> VS Code applies individual autofixes and suggestions independently of ESLint, based on the provided [`EditInfo`](https://eslint.org/docs/latest/developer-guide/nodejs-api#-editinfo-type). ESLint doesn't know if the fix/suggestion is applied in the editor. It seems that we should provide an API for editors to notify ESLint when the fix/suggestion has been applied.

Thank you for pointing this out. I hadn't realized that they didn't use an API to make the change.

The `applied` method could act as an API to ESLint to make it known that a suggestion was used.

This is useful generating stats / metrics on which suggestions are used more frequently. 

But, this changes the "scope" of this RFC beyond just ESLint to tell a plugin if a suggestion was used.

Reply:
@dbaeumer,

Do you have a suggestion on an API for letting ESLint + plugins known that a suggestion was used?